Urban-Fun/View Controllers/ActivityDetailsViewController.m

```diff
@@ -31,75 +34,86 @@ - (void)viewDidLoad {
 }
 
 - (IBAction)didTapJoin:(id)sender {
-    PFUser *currentUser = [PFUser currentUser];
-    if ([self.activity.host.objectId isEqualToString:currentUser.objectId]){
-        }
-    else{
+    User *currentUser = [User currentUser];
+    if (![self.activity.host.objectId isEqualToString:currentUser.objectId]){
         [Activity updateAttendanceListWithUserId:currentUser.objectId withActivity:self.activity withCompletion:^(BOOL succeeded, NSError * _Nullable error) {
-            if ([self.activity.attendanceList containsObject:currentUser.objectId]){
-                [self.detailsJoinButton setSelected:YES];
-            }
-            else if (![self.activity.attendanceList containsObject:currentUser.objectId]){
-                [self.detailsJoinButton setSelected:NO];
-            }
             [self setUpView];
-            
-        }];
-        
-    }
-    
+    }];
+        }
+    [self.activitydetailsDelegate syncButtons];
 }
 
 - (void)setUpView {
     self.detailsTitle.text = self.activity.title;
     self.detailsHostLabel.text = [NSString stringWithFormat:@"%@%@", @"Host - ", self.activity.host[@"username"]];
-    self.detailsAttendanceLabel.text = [NSString stringWithFormat:@"%@%lu", @"Attendance - ", (unsigned long)self.activity.attendanceList.count];
+    self.detailsAttendanceLabel.text = [NSString stringWithFormat:@"%@%lu", @"Attendance - ", self.activity.attendanceList.count];
```

Comment:
Xcode gives me a yellow warning when I use %d...is there something else I am supposed to do?

Reply:
Checked with @mbooker5 and this is the right formatter for this case, since the count is `NSUInteger`
From the documentation [here](https://developer.apple.com/documentation/foundation/nsarray/1409982-count?language=objc)
Thanks for the call out Maize!